glide-core/src/client/mod.rs

```diff
@@ -95,13 +95,89 @@ pub struct Client {
 }
 
 async fn run_with_timeout<T>(
-    timeout: Duration,
+    timeout: Option<Duration>,
     future: impl futures::Future<Output = RedisResult<T>> + Send,
 ) -> redis::RedisResult<T> {
-    tokio::time::timeout(timeout, future)
-        .await
-        .map_err(|_| io::Error::from(io::ErrorKind::TimedOut).into())
-        .and_then(|res| res)
+    match timeout {
+        Some(duration) => tokio::time::timeout(duration, future)
+            .await
+            .map_err(|_| io::Error::from(io::ErrorKind::TimedOut).into())
+            .and_then(|res| res),
+        None => future.await,
+    }
+}
+
+// Extension to the request timeout for blocking commands to ensure we won't return with timeout error before the server responded
+const BLOCKING_CMD_TIMEOUT_EXTENSION: f64 = 0.5; // seconds
```

Comment:
Why 0.5 seconds and not another value like 0.1?  I'm not sure what's best here, but this value feels somewhat arbitrary.

Reply:
We are running blocking commands with timeout only as an extra mechanism to be sure we don't hang up on the request. In the happy path, the server will return with a response up to the specified timeout. If from some other reason - the server is busy, the server is failed - we don't get any response, we will timeout the request on the client. The most important thing is not to time out before the server responded, if the server was able to respond in time. Since the round-trip of the request and the queue of other requests might take some time, we're better of setting this timeout extension a bit higher. Again, in the happy path this timeout would have no affect - the server will respond when an element is popped, or when the timeout will reached.

Reply:
Got it. Thanks for the explanation.